src/backend/parser/analyze.c

```diff
@@ -559,6 +559,11 @@ transformDeleteStmt(ParseState *pstate, DeleteStmt *stmt)
 	qual = transformWhereClause(pstate, stmt->whereClause,
 								EXPR_KIND_WHERE, "WHERE");
 
+	qry->limitCount = transformLimitClause(pstate, stmt->limitCount,
```

Comment:
Since this is a TSQL-only field, we should restrict it to be only analyzed under TSQL dialect. Consider using a hook instead to avoid more engine modification in the future